### PR TITLE
Typing annotations for VectorBus.__init__(...) are missing

### PR DESCRIPTION
**Pull request by felixdivo**
Implements #1082. Features:
- more complete docstring (parameters were missing)
- actual typing annotations
- more general type for `channel` and better parsing
- small doc formatting improvements

### CODE REVIEW

File: can/interfaces/vector/canlib.py [hunk 1]
```diff
@@ -71,59 +72,64 @@ class VectorBus(BusABC):
...
-            CAN: range 16…32768
-            CAN-FD: range 8192…524288
-        :param str app_name:
-            Name of application in Hardware Config.
-            If set to None, the channel should be a global channel index.
-        :param int serial:
+            CAN: range `16…32768`
+            CAN-FD: range `8192…524288`
+        :param app_name:
+            Name of application in the hardware config.
```
Comment (zariiii9003) [inline can/interfaces/vector/canlib.py:110]:
```suggestion
            Name of application in Vector Hardware Configuration.
```

This is the name of the Vector tool.

File: can/interfaces/vector/canlib.py [hunk 2]
```diff
@@ -71,59 +72,64 @@ class VectorBus(BusABC):
...
+            If set to `None`, the channel should be a global channel index.
+        :param serial:
             Serial number of the hardware to be used.
             If set, the channel parameter refers to the channels ONLY on the specified hardware.
-            If set, the app_name does not have to be previously defined in Vector Hardware Config.
-        :param bool fd:
+            If set, the app_name does not have to be previously defined in the Vector hardware
```
Comment (zariiii9003) [inline can/interfaces/vector/canlib.py:115]: Same here

**Review by zariiii9003 — CHANGES_REQUESTED** [review]

File: can/interfaces/vector/canlib.py [hunk 3]
```diff
@@ -71,59 +72,64 @@ class VectorBus(BusABC):
...
+            config.
```
Comment (felixdivo) [inline can/interfaces/vector/canlib.py:116]:
```suggestion
            If set, the app_name does not have to be previously defined in Vector Hardware
            Configuration.
```

**Comment by felixdivo** [comment]
@zariiii9003 Is it finished like this?

**Review by zariiii9003 — APPROVED** [review]

**Pull request merged**